monitoring/perf_context.cc

```diff
@@ -112,6 +120,45 @@ void PerfContext::Reset() {
     ss << #counter << " = " << counter << ", ";  \
   }
 
+#define PERF_CONTEXT_BY_LEVEL_OUTPUT_ONE_COUNTER(counter)         \
+  if (!exclude_zero_counters && per_level_perf_context_enabled && \
+      level_to_perf_context) {                                    \
+    ss << #counter << " = ";                                      \
+    for (auto iter = level_to_perf_context->begin();              \
+         iter != level_to_perf_context->end(); ++iter) {          \
+      if (iter->second) {                                         \
+        ss << iter->second->counter << ", ";                      \
+      }                                                           \
+    }                                                             \
+  }
+
+#define PERF_CONTEXT_BY_LEVEL_OUTPUT_FULL(level)                            \
+  if (!exclude_zero_counters && per_level_perf_context_enabled &&           \
+      level_to_perf_context && (*level_to_perf_context.contains(level)) &&  \
+      (*level_to_perf_context)[level]) {                                    \
+    ss << "all by level counters for level " << #level << " : " \
+    << "bloom_filter_useful = " \
+    << (*level_to_perf_context)[level]->bloom_filter_useful << "," \
+    << "bloom_filter_full_positive = " \
+    << (*level_to_perf_context)[level]->bloom_filter_full_positive << "," \
+    << "bloom_filter_full_true_positive = " \
+    << (*level_to_perf_context)[level]->bloom_filter_full_true_positive << ",";\
```

Comment:
If we add more counters per level, this macro will grow. It doesn't seem to be a scalable solution to me.
I don't see the necessarily of this macro. Can you use a helper function instead?